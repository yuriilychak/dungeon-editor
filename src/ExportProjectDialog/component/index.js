import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import LinearProgress from '@material-ui/core/LinearProgress';
import PropTypes from "prop-types";
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from "react-i18next";

const useStyles = makeStyles({
    button: {
        minWidth: 100
    },
    container: {
        minWidth: 400
    }
});

const ExportProjectDialog = props => {
    const { t } = useTranslation();
    const classes = useStyles();
    const contentTextId = "export-project-title";
    const { isPopupOpen, onClosePopup, onExportProject, locale, progressData } = props;
    const buttonType = "contained";
    let progress = 0;

    let progressText = "";

    if (progressData !== null) {
        progress = progressData.progress;

        if (progress !== 100) {
            progressText = t(locale.exportProgressText).replace("{0}", progressData.fileName);

        }
        else {
            if (progressData.isComplete) {
                progressText = t(locale.exportCompleteText);
            }
            else {
                progressText = t(locale.exportGenerateText);
                progress = 99;
            }
        }
    }
    else {
        progressText = t(locale.exportBeginText);
    }

    return (
        <Dialog
            open={isPopupOpen}
            aria-labelledby={contentTextId}
            classes={{
                paper: classes.container
            }}
        >
            <DialogTitle id={contentTextId}>
                { t(locale.contentTitle) }
            </DialogTitle>
            <DialogContent>
                <DialogContentText>
                    { t(locale.contentText) }
                </DialogContentText>
                <DialogContentText>
                        { progressText }
                </DialogContentText>
                <LinearProgress color="primary" variant="determinate" value={progress} />
            </DialogContent>
            <DialogActions>
                <Button
                    color="secondary"
                    variant={ buttonType }
                    className={ classes.button }
                    onClick={ onClosePopup }
                    id="exportProjectDialog-cancel"
                >
                    { t(locale.buttonCancel) }
                </Button>
                <Button
                    disabled={ progress !== 100 }
                    color="primary"
                    variant={ buttonType }
                    className={ classes.button }
                    onClick={ onExportProject }
                    id="exportProjectDialog-save"
                >
                    { t(locale.buttonSave) }
                </Button>
            </DialogActions>
        </Dialog>
    );
};

ExportProjectDialog.propTypes = {
    locale: PropTypes.shape({
        buttonCancel: PropTypes.string.isRequired,
        buttonSave: PropTypes.string.isRequired,
        contentText: PropTypes.string.isRequired,
        contentTitle: PropTypes.string.isRequired,
        exportBeginText: PropTypes.string.isRequired,
        exportCompleteText: PropTypes.string.isRequired,
        exportGenerateText: PropTypes.string.isRequired,
        exportProgressText: PropTypes.string.isRequired
    }).isRequired,
    onExportProject: PropTypes.func.isRequired,
    onClosePopup: PropTypes.func.isRequired,
    isPopupOpen: PropTypes.bool.isRequired,
    progressData: PropTypes.shape({
        progress: PropTypes.number.isRequired,
        fileName: PropTypes.string,
        isComplete: PropTypes.bool.isRequired
    })

};

export default ExportProjectDialog;
