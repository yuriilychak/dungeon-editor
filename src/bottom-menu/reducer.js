import StaticData from "./data";
import STATE from "./state";
export const initialState = {
    ...StaticData,
    sectionId: -1,
    fileId: -1,
    tabIndex: 0,
    elements: []
};

const actionHandlers = {
    [STATE.OPEN_ELEMENT]: (state, { fileId, sectionId }) => {
        return {
            ...state,
            fileId,
            sectionId,
            tabIndex: 0
        }
    },
    [STATE.CHANGE_TAB]: (state, tabIndex) => ({ ...state, tabIndex })
};

export default function topMenuReducer(state = initialState, { type, payload}) {
    const actionHandler = actionHandlers[type];
    return actionHandler ? actionHandler(state, payload) : state;
}

