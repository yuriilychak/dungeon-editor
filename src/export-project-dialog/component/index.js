export { default as ExportProjectDialog } from "./export-project-dialog";
