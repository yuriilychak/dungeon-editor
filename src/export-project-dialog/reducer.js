import STATE from "./state";
import {generateReducerData} from "../helpers";

/**
 * @typedef {Object}
 * @name ProgressData
 * @property {number} progress
 * @property {string} fileName
 * @property {boolean} isComplete
 */

/**
 * @typedef {Object}
 * @name ExportProjectDialogState
 * @property {Object.<string, string>} locale
 * @property {boolean} isPopupOpen
 * @property {?ProgressData} progressData
 */


/**
 * @param {ExportProjectDialogState} state
 * @param {ActionData} action
 * @return {ExportProjectDialogState}
 */

export default generateReducerData(
    {
        isPopupOpen: false,
        progressData: null
    },
    {
        [STATE.CHANGE_ACTIVITY]: (state, isPopupOpen) => ({
            ...state,
            isPopupOpen,
            progressData: null
        }),
        [STATE.PROGRESS]: (state, progressData) => ({ ...state, progressData })
    }
);
