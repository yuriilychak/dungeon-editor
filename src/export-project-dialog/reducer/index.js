import STATE from "../state";
import StaticData from "../data";

/**
 * @typedef {Object}
 * @name ProgressData
 * @property {number} progress
 * @property {string} fileName
 * @property {boolean} isComplete
 */

/**
 * @typedef {Object}
 * @name ExportProjectDialogState
 * @property {Object.<string, string>} locale
 * @property {boolean} isPopupOpen
 * @property {?ProgressData} progressData
 */

/**
 * @type {ExportProjectDialogState}
 */

export const initialState = {
    ...StaticData,
    isPopupOpen: false,
    progressData: null
};

/**
 * @type {Object.<string, function(ExportProjectDialogState, ActionData): ExportProjectDialogState>}
 */

const actionHandlers = {
    [STATE.CHANGE_ACTIVITY]: (state, action) => ({
        ...state,
        isPopupOpen: action.payload,
        progressData: null
    }),
    [STATE.PROGRESS]: (state, action) => ({
        ...state,
        progressData: action.payload
    })

};

/**
 * @param {ExportProjectDialogState} state
 * @param {ActionData} action
 * @return {ExportProjectDialogState}
 */

export default function topMenuReducer(state = initialState, action) {
    const actionHandler = actionHandlers[action.type];
    return actionHandler ? actionHandler(state, action) : state;
}
