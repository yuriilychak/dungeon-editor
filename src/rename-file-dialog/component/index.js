export { default as RenameFileDialog } from "./rename-file-dialog";
