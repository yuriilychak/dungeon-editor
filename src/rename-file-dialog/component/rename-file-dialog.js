import React, { useState, memo } from 'react';
import { bool, shape, string, func, number } from 'prop-types';
import { useTranslation } from "react-i18next";

import {ActionDialog} from "../../common-ui/action-dialog";
import {NameInput} from "../../common-ui/name-input";

const RenameFileDialog = props => {
    const { t } = useTranslation();
    const { staticData, isPopupOpen } = props;
    const { locale, maxNameLength } = staticData;
    const [ isError, setError ] = useState(false);

    let nameInputRef = React.createRef();

    const onCancelNewProject = () => {
        setError(false);
        props.onClosePopup();
    };

    const onSubmitNewProject = () => {
        const projectName = nameInputRef.state.projectName;

        if (projectName === "") {
            setError(true);
            return;
        }

        setError(false);
        props.onSubmitRename(projectName);
    };

    return (
        <ActionDialog
            open={isPopupOpen}
            dialogId={"rename-file-dialog"}
            title={t(locale.contentTitle)}
            description={t(locale.contentText)}
            submitTitle={t(locale.buttonSubmit)}
            rejectTitle={t(locale.buttonCancel)}
            onSubmit={onSubmitNewProject}
            onReject={onCancelNewProject}

        >
            <NameInput
                isError={ isError }
                ref={ ref => nameInputRef = ref }
                titleLocale={ t(locale.inputTitle) }
                errorLocale={t(locale.inputError)}
                maxLength={ maxNameLength }
            />
        </ActionDialog>
    );
};

RenameFileDialog.propTypes = {
    isPopupOpen: bool,
    staticData: shape({
        locale: shape({
            buttonCancel: string,
            buttonSubmit: string,
            contentTitle: string,
            contentText: string,
            inputError: string,
            inputTitle: string
        }),
        maxNameLength: number
    }),
    onClosePopup: func,
    onSubmitProject: func
};

export default memo(RenameFileDialog);
