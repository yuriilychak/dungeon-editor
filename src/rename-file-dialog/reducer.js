import STATE from "./state";
import {generateReducerData} from "../helpers";

export default generateReducerData(
    {
        isPopupOpen: false
    },
    {
        [STATE.CHANGE_ACTIVITY]: (state, isPopupOpen) => ({ ...state, isPopupOpen })
    }
);

