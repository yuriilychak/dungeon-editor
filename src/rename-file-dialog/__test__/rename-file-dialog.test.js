import { RenameFileDialog } from "../component";
import { initialState } from '../reducer';
import React from "react";
import { createMount } from '@material-ui/core/test-utils';
import "../../Locale";

describe('rename-file-dialog index test',() => {
    const props = {
        ...initialState,
        isPopupOpen: true,
        onClosePopup: jest.fn(),
        onSubmitRename: jest.fn()
    };

    const wrapper = createMount()(
        <RenameFileDialog
            {...props}
        />
    );

    const projectInput = wrapper.find('input');

    it('Open snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Close snapshot', () => {
        const wrapper = createMount()(
            <RenameFileDialog
                {...initialState}
                isPopupOpen={false}
                onClosePopup={ jest.fn() }
                onSubmitRename={ jest.fn() }
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Click submit with empty project name', () => {
        jest.spyOn(wrapper.props(), "onSubmitRename");

        const button = wrapper.find('#rename-file-dialog-submit').first();

        projectInput.simulate('change', { target: { value: '' } });
        button.simulate('click');
        expect(wrapper.props().onSubmitRename).not.toHaveBeenCalled();
    });

    it('Click submit with "test" project name', () => {
        jest.spyOn(wrapper.props(), "onSubmitRename");

        const button = wrapper.find('#rename-file-dialog-submit').first();

        projectInput.simulate('change', { target: { value: 'test' } });
        button.simulate('click');
        expect(wrapper.props().onSubmitRename).toHaveBeenCalled();
    });

    it('Click cancel', () => {
        jest.spyOn(wrapper.props(), "onClosePopup");

        const button = wrapper.find('#rename-file-dialog-reject').first();
        button.simulate('click');
        expect(wrapper.props().onClosePopup).toHaveBeenCalled();
    });
});
