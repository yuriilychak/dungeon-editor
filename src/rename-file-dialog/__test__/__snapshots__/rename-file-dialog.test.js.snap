// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rename-file-dialog index test Close snapshot 1`] = `""`;

exports[`rename-file-dialog index test Open snapshot 1`] = `""`;
